public/app/features/dashboard-scene/utils/utils.ts

```diff
@@ -275,6 +275,34 @@ export function activateSceneObjectAndParentTree(so: SceneObject): CancelActivat
   };
 }
 
+/**
+ * Adaptation of activateSceneObjectAndParentTree specific for PanelSearchLayout use case with
+ *   with panelSearch and panelsPerRow custom panel filtering logic.
+ *
+ * Activating the whole tree because dashboard does not react to variable updates such as panel repeats
+ */
+export function forceActivateFullSceneObjectTree(so: SceneObject): CancelActivationHandler | undefined {
```

Comment:
I would love to add some tests for the util, it would help to see the use case :) 

Besides that, I see the main difference from `activateSceneObjectAndParentTree` is that we stop when one node is activated. Do you think that we could just force it always? Maybe @torkelo has some context here that can help with this